dbms/src/IO/S3Common.cpp

```diff
@@ -3,58 +3,68 @@
 #include <IO/WriteBufferFromString.h>
 
 #include <iterator>
-#include <sstream>
-
-#include <Poco/Base64Encoder.h>
-#include <Poco/HMACEngine.h>
-#include <Poco/SHA1Engine.h>
-#include <Poco/URI.h>
+#include <aws/core/Aws.h>
+#include <aws/core/auth/AWSCredentialsProvider.h>
 
 
 namespace DB
 {
 
 namespace ErrorCodes
 {
-    extern const int CANNOT_FORMAT_DATETIME;
+    extern const int BAD_ARGUMENTS;
 }
 
-void S3Helper::authenticateRequest(
-    Poco::Net::HTTPRequest & request,
+
+static std::mutex aws_init_lock;
+static Aws::SDKOptions aws_options;
+static std::atomic<bool> aws_initialized(false);
```

Comment:
Replace with function level static variable (with a stuct with initialization in constructor) or ext::singleton.

Reply:
You mean place these 3 variables into a struct and declare this struct statically at function that initializes AWS API?